module/dkg/broker.go

```diff
@@ -185,8 +197,44 @@ func (b *Broker) checkMessageInstanceAndOrigin(msg messages.DKGMessage) error {
 		return fmt.Errorf("wrong DKG instance. Got %s, want %s", msg.DKGInstanceID, b.dkgInstanceID)
 	}
 	// check that the message's origin is not out of range
-	if msg.Orig >= len(b.committee) || msg.Orig < 0 {
+	if msg.Orig >= uint64(len(b.committee)) {
 		return fmt.Errorf("origin id out of range: %d", msg.Orig)
 	}
 	return nil
 }
+
+// prepareBroadcastMessage creates BroadcastDKGMessage with a signature from the
+// node's staking key.
+func (b *Broker) prepareBroadcastMessage(data []byte) (messages.BroadcastDKGMessage, error) {
+	dkgMessage := messages.NewDKGMessage(
+		b.myIndex,
+		data,
+		b.dkgInstanceID,
+	)
+	sigData := flow.MakeID(dkgMessage)
+	signature, err := b.me.Sign(sigData[:], hasher.NewDKGMessageHasher())
```

Comment:
`MakeID` hashes a serialization of the structure, which is not required before signing. We should just sign the serialization of the structure using https://github.com/onflow/flow-go/blob/master/model/fingerprint/fingerprint.go#L21. 
We can work with the RLP serialization for now.